include/libtorrent/torrent_info.hpp

```diff
@@ -342,6 +342,8 @@ namespace libtorrent {
 			TORRENT_ASSERT(m_files.num_pieces() > 0);
 			return piece_index_t(m_files.num_pieces());
 		}
+		aux::index_range<piece_index_t> piece_range() const
```

Comment:
I think this leak `aux::` to the public API

Reply:
I would like the return type to be an *unspecified type* that can be used in a range-for loop. But my simple documentation generation script can't hide such concrete types.

Reply:
I think that if this `index_range` is intended to be public API, it should be outside `aux`

Reply:
well, not sure if "unspecified" has some special semantic here (modern c++ has a lot of them lately :), but maybe `using piece_range_t = aux::index_range<piece_index_t>` inside `torrent_info` do the trick of not having `aux::` in public API.

Reply:
Is there some practical (or otherwise) benefit of returning an alias for an internal type over the internal type itself?
The thing I can think of is that the `aux::index_range` becomes part of the ABI here, and it living in the `aux` namespace may be misleading to make it look like it doesn't affect library ABI.

Reply:
yes, I would say that's the only reason, the `aux::` happening in public ABI (at first I was concerned only about API)

Reply:
I moved `index_range` into its own header and out of the `aux` namespace